custom_connectors/oauth2/trackvia.rb

```diff
@@ -807,181 +766,232 @@
+              error("#{message} : #{body}")
+            end&.presence || 'success'
+        }
+      end,
+
+      output_fields: lambda do |_object_definitions|
+        [{ name: 'status' }]
+      end,
+
+      sample_output: lambda do |_connection, _input|
+        { status: 'success' }
+      end
+    }
+  },
+
+  triggers: {
+    new_record: {
+      description: "New <span class='provider'>record</span> added to view in <span class='provider'>TrackVia</span>",
+      help: 'Fetch records added to a TrackVia view in real-time. Records added when the trigger is stopped will not be processed.',
+      config_fields: [
+        {
+          name: 'app_name',
+          label: 'Application',
+          control_type: 'select',
+          pick_list: 'apps',
+          optional: false,
+          hint: 'Select a TrackVia application.'
+        },
+        {
+          name: 'view_id',
+          label: 'View',
+          control_type: 'select',
+          pick_list: 'views',
+          pick_list_params: { app_name: 'app_name' },
+          optional: false,
+          hint: 'Select an available view.',
+          toggle_hint: 'Select from list',
+          toggle_field: {
+            name: 'view_id',
+            label: 'View ID',
+            type: :integer,
+            disable_formula: true,
+            control_type: 'plain_text',
+            toggle_hint: 'Enter custom value',
+            hint: 'Select the required view. ID can be found at the end of URL.'
+          }
+        }
+      ],
+
+      webhook_notification: lambda do |_input, payload, _e_i_s, e_o_s|
+        payload.map do |record|
+          record.each do |key, val|
+            record[key] = val.map { |value| { 'choice' => value } } if val.is_a?(Array) && val.first&.is_a?(String)
           end
-        end,
+          e_o_s.each_with_object({}) do |k, hash|
+            hash[k[:name]] = record[k[:label]]
+          end.compact.merge('id' => record['id'])
+        end
+      end,
 
-        webhook_subscribe: lambda do |webhook_url, _connection, input|
-          post("zapier/views/#{input['view_id']}/api/hooks",
-               target_url: webhook_url,
-               event: 'created').merge(view_id: input['view_id'])
-        end,
+      webhook_subscribe: lambda do |webhook_url, _connection, input|
+        post("zapier/views/#{input['view_id']}/api/hooks",
+             target_url: webhook_url,
+             event: 'created')
+          .merge(view_id: input['view_id'])
+      end,
 
-        webhook_unsubscribe: lambda do |webhook|
-          delete("zapier/views/#{webhook['view_id']}/api/hooks/#{webhook['id']}")
-        end,
+      webhook_unsubscribe: lambda do |webhook|
+        delete("zapier/views/#{webhook['view_id']}/api/hooks/#{webhook['id']}")
+      end,
 
-        dedup: lambda do |record|
-          record['id']
-        end,
+      dedup: lambda do |record|
+        "#{record['id']}@#{record['dedup_modifiedDate']}"
+      end,
```

Comment:
Why dedup on modified date for new records?

Reply:
Trackvia is capable of sending the same record multiple times as part of its webhook.  We had an instance of the same recording coming through and triggering workato actions when it had already been processed.  Adding the modified date should only allow records that have been changed to be polled